platform/engineering/design-docs/2020-12-10-auto-deploy-cms-build.md

```diff
@@ -0,0 +1,220 @@
+
+For convenience, we could also upload the checksum to `s3://vetsgov-website-builds-s3-upload/cms-export/data/checksum.txt`. With that, the pipeline would not have to download the full tarball for the comparison.
+
+### Code Location
+
+The pipeline will be defined in `Jenkinsfile.autodeploy`. The [helper functions](https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy) that it uses are in `jenkins/common.groovy`.
+
+There may be auxiliary configurations required in the `devops` [repo](https://github.com/department-of-veterans-affairs/devops), likely in the `ansible` directory.
+
+### Testing Plan
+
+The following behaviors should be observed cumulatively as they are implemented.
+1. Build is stored at the proper location in S3 when the pipeline is invoked.
+2. Pipeline runs on the defined schedule.
+3. Deploy doesn't proceed when no new content has been published.
+
+### Logging
+
+On top of the existing build logs, the new content-only deploy will output logs for the new transformation processes in the CMS export build.
+
+### Debugging
+
+The auto-deploy pipeline can be debugged by reading its console output. The S3 bucket can be checked to confirm whether the build has been deployed.
+
+The CMS export build will validate the input and output schemas of the transformers, so errors related to the build itself can be debugged using that output.
+
+As with the full build, during the rollout, there will be a comparison of GraphQL and CMS export builds to report errors when discrepancies between the two are found. The script used for this comparison will have some console output as well as an output JSON file describing the discrepancies.
+
+To determine whether a page exists in the Drupal content, we can go to [`staging.va.gov/drupal/debug/`](staging.va.gov/drupal/debug/). This page is not available in production, however.
+
+### Caveats
+
+To be determined.
+
+### Security Concerns
+
+There are no new security concerns with the auto-deploy. There is no user interaction involved unless we include a manual trigger for use by internal VSP engineers, which still should not present any threat of malicious requests. Denial of service is also not a concern as builds are queued.
+
+### Privacy Concerns
+
+There are no new privacy concerns with the auto-deploy as user data is not involved.
+
+### Open Questions and Risks
+
+#### Questions
+
+- What is the ideal schedule for the auto-deploy?
+  - The pipeline will initially run hourly 8am-8pm ET on weekdays.
+  - We want the highest rate of deploy that the system can sustain.
+
+- How should we coordinate the schedule of the auto-deploy with the daily production deploy?
+  - The deploys from both pipelines will likely overlap.
+  - The daily production deploy is still necessary for tagging the release.
```

Comment:
Would it be better to move the 'release' job to a separate job that doesn't then trigger the deploy?

Reply:
I don't find the coupling of the release and deploy to be a detriment right now, so I'm not sure if it would be necessary for the release. But having the separation would be cleaner and should probably be considered in the move to CircleCI.